storage/pruning/pruningStorer.go

```diff
@@ -364,6 +365,77 @@ func (ps *PruningStorer) GetFromEpoch(key []byte, epoch uint32) ([]byte, error)
 
 }
 
+// GetBulkFromEpoch will return a slice of keys only in the persister for the given epoch
+func (ps *PruningStorer) GetBulkFromEpoch(keys [][]byte, epoch uint32) (map[string][]byte, error) {
+	ps.lock.RLock()
+	pd, exists := ps.persistersMapByEpoch[epoch]
+	ps.lock.RUnlock()
+	if !exists {
+		log.Warn("get from removed persister",
+			"id", ps.identifier,
+			"epoch", epoch)
+		return nil, errors.New("persister does not exits")
+	}
+
+	var persisterForEpoch storage.Persister
+	var err error
+
+	if pd.isClosed {
```

Comment:
@AdoAdoAdo  Please confirm this logic is correct.

- If persister data is closed, open it, initialize it, close it at the end of the function